File: testdata/jennies/builders/map_of_disjunctions/PHPBuilder/src/MapOfDisjunctions/ElementBuilder.php
<?php

namespace Grafana\Foundation\MapOfDisjunctions;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\Element>
 */
class ElementBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\MapOfDisjunctions\Element $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\MapOfDisjunctions\Element();
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\MapOfDisjunctions\Element
     */
    public function build()
    {
        return $this->internal;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\Panel> $panel
     */
    public function panel(\Grafana\Foundation\Cog\Builder $panel): static
    {
        $panelResource = $panel->build();
        $this->internal->panel = $panelResource;
    
        return $this;
    }

    /**
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\MapOfDisjunctions\LibraryPanel> $libraryPanel
     */
    public function libraryPanel(\Grafana\Foundation\Cog\Builder $libraryPanel): static
    {
        $libraryPanelResource = $libraryPanel->build();
        $this->internal->libraryPanel = $libraryPanelResource;
    
        return $this;
    }

}

File: testdata/jennies/builders/map_of_disjunctions/JavaBuilder/map_of_disjunctions/ElementBuilder.java
package map_of_disjunctions;


public class ElementBuilder implements cog.Builder<Element> {
    protected final Element internal;
    
    public ElementBuilder() {
        this.internal = new Element();
    }
    public ElementBuilder panel(cog.Builder<Panel> panel) {
    Panel panelResource = panel.build();
        this.internal.panel = panelResource;
        return this;
    }
    
    public ElementBuilder libraryPanel(cog.Builder<LibraryPanel> libraryPanel) {
    LibraryPanel libraryPanelResource = libraryPanel.build();
        this.internal.libraryPanel = libraryPanelResource;
        return this;
    }
    public Element build() {
        return this.internal;
    }
}
